### PR TITLE
Implement GraphQL edge relationships with thunk-style type references

### PR DESCRIPTION
**Pull request by randallb**
Add support for thunk-style type references to solve circular dependency issues in GraphQL edge relationships. This implementation simplifies the API by making the field name serve as the edge role and making edge relationships implicit for object fields without custom resolvers.

Changes:
- Update makeGqlBuilder.ts with thunk-style type references for object fields
- Implement edge relationship resolver in gqlSpecToNexus.ts
- Update BfPerson.ts to use dynamic imports for circular dependency resolution
- Add comprehensive documentation and examples in implementation plan
- Update project status to reflect completed edge relationship implementation
- Add tests for edge relationship specification and configuration

Test plan:
1. All types pass type checking (bff check apps/bfDb/builders/graphql)
2. All tests pass (bff test apps/bfDb/builders/graphql)
3. Ensure edge relationship API is well-documented

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>
---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/bolt-foundry/bolt-foundry/pull/761).
* #762
* __->__ #761
* #759

### CODE REVIEW

File: apps/bfDb/builders/graphql/gqlSpecToNexus.ts [hunk 1]
```diff
@@ -10,13 +10,25 @@
...
 
 /**
- * Type definition for an edge relationship spec
+ * Type definition for an edge relationship specification
+ *
+ * This defines how an edge relationship is represented in the GraphQL schema
+ * and provides the information needed to resolve the relationship at runtime.
  */
 type EdgeRelationshipSpec = {
+  /** GraphQL type name (initially same as field name) */
```
Comment (randallb) [inline apps/bfDb/builders/graphql/gqlSpecToNexus.ts:19]: maybe we should derive it? Like PersonMemberOfOrganization?

File: apps/bfDb/builders/graphql/gqlSpecToNexus.ts [hunk 2]
```diff
@@ -10,13 +10,25 @@
...
   type: string;
+
+  /** GraphQL field arguments */
   args: Record<string, unknown>;
```
Comment (randallb) [inline apps/bfDb/builders/graphql/gqlSpecToNexus.ts:23]: I think there's a GraphlqlInput type or something.

File: apps/bfDb/builders/graphql/gqlSpecToNexus.ts [hunk 3]
```diff
@@ -10,13 +10,25 @@
...
-  edgeRole: string; // The field name that serves as the role
+
+  /** The role name for the edge (same as field name in the current implementation) */
+  edgeRole: string;
```
Comment (randallb) [inline apps/bfDb/builders/graphql/gqlSpecToNexus.ts:26]: We should remove this.

File: apps/bfDb/builders/graphql/gqlSpecToNexus.ts [hunk 4]
```diff
@@ -106,19 +118,36 @@ function createDefaultRelationResolver(relationName: string) {
...
+ *
+ * This resolver handles the runtime resolution of relationships between nodes using BfEdge.
+ * It follows these steps:
+ * 1. Queries BfEdge with the source node ID and edge role (e.g., "memberOf")
+ * 2. Extracts the target node information from the first matching edge
+ * 3. Loads and returns the target node
+ *
  * @param relationName The name of the field/relationship (e.g., "memberOf")
- * @param edgeRole The role name for the edge (same as relationName)
+ * @param edgeRole The role name for the edge (same as relationName in current impl)
```
Comment (randallb) [inline apps/bfDb/builders/graphql/gqlSpecToNexus.ts:129]: We probably want to just have a prop on a BfEdge called "BfRelationshipName" or something.

File: apps/bfDb/builders/graphql/gqlSpecToNexus.ts [hunk 5]
```diff
@@ -179,11 +211,11 @@ function createEdgeRelationshipResolver(
...
         );
-        
+
         // Due to circular dependencies between BfNode and BfEdge, we need to use a type assertion
         // deno-lint-ignore no-explicit-any
         type BfEdgeQuery = any; // This is a simplification for type safety
-        
+
         // Call the query method with proper parameters
         const edges = await (BfEdgeModule.BfEdge as BfEdgeQuery).query(
```
Comment (randallb) [inline apps/bfDb/builders/graphql/gqlSpecToNexus.ts:220]: Double check that this is right... it might be wrong.

File: apps/bfDb/builders/graphql/gqlSpecToNexus.ts [hunk 6]
```diff
@@ -357,12 +388,12 @@ export function gqlSpecToNexus(spec: GqlNodeSpec, typeName: string) {
...
           // Add resolver based on relationship type with debug wrapper
-          resolve: async function(
+          resolve: async function (
             // deno-lint-ignore no-explicit-any
-            root: any, 
-            args: Record<string, unknown>, 
-            ctx: BfGraphqlContext, 
-            info: GraphQLResolveInfo
+            root: any,
+            args: Record<string, unknown>,
```
Comment (randallb) [inline apps/bfDb/builders/graphql/gqlSpecToNexus.ts:394]: this might be able to be GraphqlInputArgs or something that nexus provides... i think the args come from graphql though.

File: apps/bfDb/builders/graphql/makeGqlBuilder.ts [hunk 7]
```diff
@@ -12,12 +12,16 @@ type ThisNode = InstanceType<AnyBfNodeCtor>;
...
- * deno-lint-ignore no-explicit-any
+ *
+ * The returned value can be:
+ * 1. A concrete class constructor (BfOrganization)
+ * 2. An abstract class constructor (BfNode)
+ * 3. A module promise with a named export (.then(m => m.BfOrganization))
  */
-type GqlObjectThunk = () => MaybePromise<any>; // Using any here as we need to support both concrete and abstract constructor types
+// deno-lint-ignore no-explicit-any
+type GqlObjectThunk = () => MaybePromise<any>; // Any is necessary to support various constructor forms
```
Comment (randallb) [inline apps/bfDb/builders/graphql/makeGqlBuilder.ts:24]: it might be able to be AnyGraphqlObjectBaseCtor (might have to create, but it'd be like AnyBfNodeCtor)
